### PR TITLE
Remove public TestNet, RegTest and GetNetwork from NodeSettings (Refactor)

### PR DESCRIPTION
**Pull request by quantumagi**
This PR removes unnecessary public methods and variables from `NodeSettings`.

Having the public / member `TestNet` and `RegTest` variables can allow contradictions to arise between the values of the variables and the current `this.Network` variable.

### CODE REVIEW

**Comment by bokobza** [comment]
I'll check this one.

File: src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ public void BuildWithInitialServicesSetupConfiguresFullNodeUsingConfiguration()
             this.fullNodeBuilder.ConfigureServiceProvider(e =>
             {
                 var settings = e.GetService<NodeSettings>();
-                settings.Testnet = true;
+                settings.Agent = "abc";
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs]: Why is this needed?

File: src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -36,17 +36,16 @@ public void UseNodeSettingsConfiguresNodeBuilderWithNodeSettings()
...
-            Assert.Equal(NodeSettings.Default().GetNetwork(), this.fullNodeBuilder.Network);
+            Assert.Equal(NodeSettings.Default().Network, this.fullNodeBuilder.Network);
             Assert.Single(this.serviceCollectionDelegates);
         }
 
         [Fact]
         public void UseDefaultNodeSettingsConfiguresNodeBuilderWithDefaultSettings()
         {
-            var nodeSettings = NodeSettings.Default();
+            var nodeSettings = new NodeSettings();
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderExtensionsTest.cs]: Why the change? You could have `NodeSettings.Default(Network.TestNet);`

File: src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs]: Going by the legacy code one of the purposes of the test case seems to be to prove that the node settings can be changed. Since we no longer have a Testnet variable I exchanged that for a change to the Agent variable.

File: src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderExtensionsTest.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderExtensionsTest.cs]: I was making some changes to the arguments and then settled on not changing anything. I can revert that to the Default method although it would have the same effect. Using Network.Testnet would make this test case fail due to the `Assert.Equal(Network.Main, this.fullNodeBuilder.Network);`.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderExtensionsTest.cs]: I will change all 3 methods in that class to use the Default method.

File: src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs (hunk 1 above)
Comment (bokobza) [reply src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs]: yeah please, and in BuildConfiguresFullNodeUsingConfiguration as well.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Tests/Builder/FullNodeBuilderTest.cs]: Done.

**Comment by bokobza** [comment]
Good stuff! Thank you
